From: Alex Merry <alex.merry@kde.org>
Date: Fri, 04 Jul 2014 10:10:27 +0000
Subject: Make FindGettext compatible with the one provided by CMake
X-Git-Url: http://quickgit.kde.org/?p=kdelibs4support.git&a=commitdiff&h=581ee6b370d8b5627196c8d6d8760d3655f6daea
---
Make FindGettext compatible with the one provided by CMake

This version will accept the old GETTEXT_PROCESS_PO_FILES() syntax (no
PO_FILES argument), but will also accept the new syntax required by
CMake's version of this file. It will also warn when PO_FILES is not
given.

REVIEW: 119111
---


--- a/cmake/modules/FindGettext.cmake
+++ b/cmake/modules/FindGettext.cmake
@@ -212,7 +212,16 @@
       LIST(REMOVE_AT _args 0 1)
    ENDIF("${_tmp}" STREQUAL "INSTALL_DESTINATION")
    
-#    message(STATUS "2 all ${_addToAll} dest ${_installDest} args: ${_args}")
+   LIST(GET _args 0 _tmp)
+   IF("${_tmp}" STREQUAL "PO_FILES")
+      LIST(REMOVE_AT _args 0)
+   ELSE()
+      SET(_deprecation_arg DEPRECATION)
+      IF(CMAKE_VERSION VERSION_LESS 3.0.0)
+         SET(_deprecation_arg AUTHOR_WARNING)
+      ENDIF()
+      MESSAGE(${_deprecation_arg} "GETTEXT_PROCESS_PO_FILES without a PO_FILES argument is deprecated")
+   ENDIF()
    
    FOREACH(_current_PO_FILE ${_args})
       GET_FILENAME_COMPONENT(_name ${_current_PO_FILE} NAME)

